### ISSUE TITLE
Star run_peakbag() cryptic warning

### ISSUE THREAD

**Issue by alexlyttle** [grd349/PBjam#190]
A user warning and error appeared when running the following:

```python
star.run_peakbag()
```
 Warning:

```
/Users/ajl573/.virtualenvs/base/lib/python3.7/site-packages/pbjam/peakbag.py:152: UserWarning: Did not find optimal rung location
  warnings.warn('Did not find optimal rung location')
```

Followed by an error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<timed eval> in <module>

~/.virtualenvs/base/lib/python3.7/site-packages/pbjam/star.py in run_peakbag(self, model_type, tune, nthreads, make_plots, store_chains)
    163         print('Starting peakbagging run')
    164         # Init
--> 165         self.peakbag = peakbag(self, self.asy_fit)
    166 
    167         # Call

~/.virtualenvs/base/lib/python3.7/site-packages/pbjam/peakbag.py in __init__(self, starinst, asyinst, init, path)
     67         if init:
     68             self.make_start()
---> 69             self.trim_ladder()
     70         self.gp0 = [] # Used for gp linewidth info.
     71 

~/.virtualenvs/base/lib/python3.7/site-packages/pbjam/peakbag.py in trim_ladder(self, lw_fac, extra, verbose)
    155                 print(f'w/2 = {int(w/2)}')
    156             ladder_trim_f[idx, :] = \
--> 157                 self.f[loc_mid_02 - int(w/2): loc_mid_02 - int(w/2) + int(w)]
    158             ladder_trim_s[idx, :] = \
    159                 self.s[loc_mid_02 - int(w/2): loc_mid_02 - int(w/2) + int(w) ]

ValueError: could not broadcast input array from shape (0) into shape (125)
```

`run_asy_peakbag` performed very well and the oscillilation spectrum has high SNR, it looked to be an easy fit.

**Comment by grd349** [comment]
Thanks @alexlyttle.

I'll take a look - it'll be a problem with the ladder function and failing to arrange the data as needed. 

What's the id for this star?

**Comment by alexlyttle** [comment]
Thanks Guy,

It’s a really good star:
350585437

**Comment by alexlyttle** [comment]
Same error on TIC 375089600 which is a less good spectrum. I couldn't find any obvious similarities with the two cases.

**Comment by grd349** [comment]
I also got this on ... kic = '6508366' 

This looks like a meaningful bug - need to sort.

**Comment by grd349** [comment]
Discussing with Martin we think this might be a problem with the way we are interacting with the data.  Both @alexlyttle and I reduce the periodogram range as part of our data prep.  @nielsenmb does not do this this step.

We think perhaps the failures come when the ladder function looks for a part of the periodogram that has been trimmed out.  It turns out that star takes care of the periodogram range - I had forgotten this.  MArtin has not been seeing the issue here.  This error should then go away when `star` uses the information on norders to correctly select the range of the periodogram.

So, to conclude ... don't select down the periodogram frequency range when using the `star` class (or the session for that matter).  

I will update the examples (from docs: star & session) at some point to reflect this new best practice.

**Comment by grd349** [comment]
@alexlyttle - could you see if this solution fixes your examples (and also check I am making any sense).

**Comment by alexlyttle** [comment]
I will check this. If understand what you mean, it is probably looking for modes outside of our selection. I think I only selected a range to encourage lightkurve’s seismology module to work (sometimes it looks in the wrong part of the spectrum if the `window_length` is wrong). I’ll let you know how it goes.

**Comment by alexlyttle** [comment]
When I revisited TIC 350585437 and fed in the whole spectrum by passing,

```python
pg = lc.to_periodogram(normalization='psd').flatten()
```

`star.run_peakbag()` worked! I will check the other stars this was a problem on but the above solution makes sense.

Note, I initially passed a periodogram without flattening it as the PBjam docs say the power may be in arbitrary units, but this failed. I have logged this in Issue #197
